### PR TITLE
FreeBSD and Mac does not provide abstract sockets.

### PR DESCRIPTION
**Pull request by l1gi**
My approach. Tested just on FreeBSD. Need Mac as well as Linux testing.

### CODE REVIEW

File: src/modes/thread_view/page_client.cc [hunk 1]
```diff
@@ -116,14 +116,18 @@ namespace Astroid {
...
+    socket_addr = ustring::compose ("%1/astroid.%2.%3.%4",
+        astroid->standard_paths ().socket_dir.c_str(),
         getpid(),
         id,
         UstringUtils::random_alphanumeric (30));
 
     refptr<Gio::UnixSocketAddress> addr = Gio::UnixSocketAddress::create (socket_addr,
+# if defined __FreeBSD__ || defined __APPLE__
+        Gio::UNIX_SOCKET_ADDRESS_PATH);
+# elif
```
Comment (gauteh) [inline src/modes/thread_view/page_client.cc]: should be # else

File: src/modes/thread_view/webextension/tvextension.cc [hunk 2]
```diff
@@ -147,7 +147,11 @@ AstroidExtension::AstroidExtension (
   const char * caddr = g_variant_get_string ((GVariant *) gaddr, &sz);
 
   refptr<Gio::UnixSocketAddress> addr = Gio::UnixSocketAddress::create (caddr,
+# if defined __FreeBSD__ || defined __APPLE__
+      Gio::UNIX_SOCKET_ADDRESS_PATH);
+# elif
```
Comment (gauteh) [inline src/modes/thread_view/webextension/tvextension.cc]: Same

File: src/modes/thread_view/thread_view.cc [hunk 3]
```diff
@@ -1385,12 +1385,12 @@ namespace Astroid {
 
     keys.register_key ("a",
         "thread_view.archive_thread",
-        "Toggle 'inbox' tag on the whole thread",
+        "Toggle 'archived' tag on the whole thread",
         [&] (Key) {
 
           if (!edit_mode && focused_message) {
             if (thread) {
-              main_window->actions->doit (refptr<Action>(new ToggleAction(thread, "inbox")));
```
Comment (gauteh) [inline src/modes/thread_view/thread_view.cc]: This seems unrelated, did it sneak in?

**Comment by gauteh** [comment]
I think the way of determining whether the abstract sockets is possible done in #649 is more general, though I like the path management that you do here.

File: src/modes/thread_view/thread_view.cc (hunk 3 above)
Comment (l1gi) [reply src/modes/thread_view/thread_view.cc]: Yes, sorry, I will patch it out.

**Comment by l1gi** [comment]
Now it could be complete. Could you review it, please?

File: src/config.cc [hunk 4]
```diff
@@ -338,6 +341,11 @@ namespace Astroid {
       create_directories (std_paths.runtime_dir);
     }
 
+    if (!is_directory(std_paths.socket_dir)) {
+      LOG (warn) << "cf: making socket dir..";
```
Comment (gauteh) [inline src/config.cc]: Should probably be debug level

**Comment by l1gi** [comment]
Now the permissions should be correct:

ligi@wand:~/workspace/astroid$ ls -ld ~/.cache/astroid/socket
drwx------  2 ligi  ligi  3 Sep  2 09:54 /home/ligi/.cache/astroid/socket
ligi@wand:~/workspace/astroid$ ls -l ~/.cache/astroid/socket
total 1
srwx------  1 ligi  ligi  0 Sep  2 09:54 astroid.17482.1.9z3wdebt6f7sxdvi6tmfrwal9nvvmg

I would keep warn level if you don't mind as all other LOG() calls about 'making' are warn as well. Any objection?

**Comment by gauteh** [comment]
Makes sense.

**Comment by serverhiccups** [comment]
This looks like this supersedes my pr #649 so I'll close mine.

**Comment by gauteh** [comment]
Thanks, merged.

**Comment by gauteh** [comment]
Hi, there are some issues which seem to be related to these changes: #661.
Do you have the chance to investigate?

ons. 4. sep. 2019, 07:12 skrev serverhiccups <notifications@github.com>:

**Comment by l1gi** [comment]
Hi Gaute,

unfortunately, I don't have a chance to try on different platform than 
FreeBSD.

Could be that socket bind() is called more that once without previous 
unbinding/closing?

Also, could anyone on Linux verify that proper code path is taken when 
socket type is selected, please?

Sorry for more questions than answers.

Regards
-- 
m.

**Comment by gauteh** [comment]
Hi, unfortunately I do not have time to look into this in the immediate future, so the alternative is to revert the relevant commits for the time being. If you find time to handle this in some way, then that would very good, but I understand if you do not.

File: src/modes/thread_view/page_client.cc [hunk 5]
```diff
@@ -116,14 +116,20 @@ namespace Astroid {
...
-        astroid->standard_paths ().runtime_dir.c_str(),
-        getpid(),
-        id,
-        UstringUtils::random_alphanumeric (30));
-
-    refptr<Gio::UnixSocketAddress> addr = Gio::UnixSocketAddress::create (socket_addr,
-        Gio::UNIX_SOCKET_ADDRESS_ABSTRACT);
+    refptr<Gio::UnixSocketAddress> addr;
+    if(Gio::UnixSocketAddress::abstract_names_supported ()) {
+      addr = Gio::UnixSocketAddress::create (socket_addr,
```
Comment (kyrias) [inline src/modes/thread_view/page_client.cc]: #661 is caused by `socket_addr` being uninitialized here.
